components/developer/golang-16/Makefile

```diff
@@ -0,0 +1,166 @@
+COMPONENT_INSTALL_ENV += GOPATH="$(SUPP_ROOT)"
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/justmake.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+# We have to fiddle with the package archives to get them into the same source
+# directory so that we can use the standard build targets.  We're effectively
+# building multiple components using a single target.
+COMPONENT_POST_UNPACK_ACTION= \
+	$(MV) go $(COMPONENT_SRC)
+
+COMPONENT_POST_UNPACK_ACTION_1= \
+	$(MV) go $(COMPONENT_SRC_1) && \
+	$(CP) -r $(COMPONENT_SRC_1) $(COMPONENT_SRC)/$(COMPONENT_SRC_1)
+
+# Fixup compressed data affected by patches.
+COMPONENT_PREP_ACTION= \
+	cd $(COMPONENT_SRC) && \
+	bzip2 -c src/compress/testdata/Mark.Twain-Tom.Sawyer.txt > \
+		src/compress/bzip2/testdata/Mark.Twain-Tom.Sawyer.txt.bz2
+
+# Build the bootstrap compiler before attempting to build Go 1.5+.  For the
+# bootstrap compiler ensure GOROOT_FINAL is unset as it will be executed out of
+# the build area, and ensure GOROOT_BOOTSTRAP is also not set.  GOEXPERIMENT
+# must also be unset as it is 1.5+ only.
+COMPONENT_PRE_BUILD_ACTION= \
+	cd $(@D)/$(COMPONENT_SRC_NAME_1)/src; \
+	$(ENV) $(COMPONENT_BUILD_ENV) \
+	GOEXPERIMENT= GOROOT_BOOTSTRAP= GOROOT_FINAL= ./make.bash
+
+# Go doesn't have a standard Makefile-based build process, so override the
+# default.
+COMPONENT_BUILD_ACTION=	\
+	cd $(@D)/src; $(ENV) $(COMPONENT_BUILD_ENV) ./make.bash
+
+COMPONENT_TARGET_DIR= \
+	$(PROTOUSRLIBDIR)/$(COMPONENT_NAME)/$(COMPONENT_MAJOR_VERSION)
+
+# We can't do any ASLR'ing as the Go 6l linker has no understanding of the
+# -z aslr=... option. So just override where needed in the .p5m files.
+ASLR_MODE = $(ASLR_DISABLE)
+
+# Make sure the source code and object files have identical timestamps so
+# the Go compiler doesn't try to rebuild them.
+PKG_MACROS += TIMESTAMP=$(shell date +"%Y%m%dT%H%M%SZ")
+# Extra macros for packaging convenience.
+PKG_MACROS += COMPONENT_MAJOR_VERSION=$(COMPONENT_MAJOR_VERSION)
+# COMPONENT_MAJOR_VERSION suitable for use in regular expressions.
+PKG_MACROS += COMPONENT_RE_MAJOR_VERSION=$(subst .,\\.,$(COMPONENT_MAJOR_VERSION))
+
+COMPONENT_TEST_ENV +=	PATH="$(@D)/bin:$(PATH)"
+COMPONENT_TEST_ENV +=	GOROOT="$(@D)"
+COMPONENT_TEST_DIR=	$(@D)/src
+COMPONENT_TEST_CMD=	./run.bash
+COMPONENT_TEST_TARGETS=
+
+define COMPONENT_INSTALL_ACTION=
+	($(MKDIR) $(COMPONENT_TARGET_DIR)/src); \
+	(cd $(SOURCE_DIR) && \
+	/usr/gnu/bin/find . -maxdepth 1 -path "./$(COMPONENT_SRC_NAME_1)" \
+	  -prune -o \( ! -iname ".*" \) -print \
+	  -exec /usr/gnu/bin/cp -r \
+	  --parents {} $(COMPONENT_TARGET_DIR) \;); \
+	(cd $(BUILD_DIR_64) && \
+	/usr/gnu/bin/find src -type f \
+		-exec /usr/gnu/bin/cp --parents {} $(COMPONENT_TARGET_DIR) \;); \
+	(cd $(BUILD_DIR_64) && $(CP) -r bin $(COMPONENT_TARGET_DIR)/); \
+	(cd $(BUILD_DIR_64) && $(CP) -r pkg $(COMPONENT_TARGET_DIR)/); \
+	($(MKDIR) $(PROTOUSRSHAREMAN1DIR));
+endef
+
+# common targets
+build:		$(BUILD_64)
```

Comment:
stock syscall is almost empty, can we define `mksysnum_solaris.pl` or other way to populate `src/syscall/zsysnum_solaris_amd64.go`?

Having this empty makes some subset of tools barely portable.

The following ruby script gives the content we need:

```ruby
puts(<<HEADER)
// +build amd64,solaris

package syscall

const (
HEADER

syscall_h = File.readlines('/usr/include/sys/syscall.h')
calls = {}
syscall_h.each do |line|
  m = line.match(/#define\s(SYS_[^\s]*)\s+(\d+)/)
  if m
    calls[m[1].upcase] = m[2]
  end
end
width = calls.keys.max_by { |x| x.length }.length + 1
calls.each do |call, num|
  printf("\t%-#{width}s = %d\n", call, num)
end
puts ")"
```